proposals/Symbiosis Protocol.md

```diff
@@ -0,0 +1,393 @@
+- 3Box maintains own infrastructure **(TODO: Verify)**
+    1. To replicate user data.
+    2. Facilitate pubsub.
+
+## HTTP API provides inadequate solution
+
+[IPFS HTTP API][] provides inadequate solution to the constrained environments
+
+
+-  It gives client an absolute control of the host node
+   - This is inpractical even with a localhost as clients (apps) need to be **isolated** and their access needs to be **revocable**.
+
+     - Browser with embeded IPFS node would need to provide each app restricted & isolated access to the node
+     - An OS level IPFS service would need to implement similar sandboxing to prevent data races.
+     - If malicious web app tricking user into giving access, assume full control of node and all the user data in it.
+     - Well intentioned clients can run into race conditions
+         - Unintentionally overwrite path in MFS.
+         - Unpin content other clients needs.
+
+- It provides inpracitcal path to incremental enhancements.
+  - By default requests from web browsers are blocked (Due to [CORS][] vialoation) which requires:
+      1. Stopping a host node.
+      2. Writing site origin into IPFS config file.
+      3. Starting a host node back up.
+
+    _In the process host will drop all other clients. They will need to keep trying to reconnect and on success recreate local state_
+    
+- Fallback mechanism (like [ipfs-provider][]) will needs to switch between JS-IPFS node and IPFS Client. This results in an observable difference:
+    - IPFS Node with different peerid / address(es).
+    - Different states of (M)File System and Pinset.
+  
+   Making it impractical when building competitive product experience.
+    
+- Not a practical option on mobile
+    - It is not viable to run a local IPFS host node on mobile device, due to backgrounding.
+    - Remote IPFS host node would require
+        - SSL/TLS certificates
+        - Some form of authentification
+        - A host per modile device.
+
+Similar problems are manifested in desktop operating systems
+- Applications may want to use different node configurations
+  > This was pressing issue when we had IPFS Desktop, Textile Desktop Photos and Radicle.
+  > They all end up embedding differnt IPFS nodes to avoid this problem.
+  > // **TODO:** Check how does [IPFS Desktop], and [Space Daemon][] get along with one another
+- Using [IPFS HTTP API][] for sharing would still inherit all the issues.
+
+## Symbiosys protocol
+
+> Protocol draws it's inspired from [DNS Service Discovery (DNS-SD)][DNS-SD] a [zero-configuration networking][] technique.
+
+Wire protocol that is agnostic of transport and representation (message encoding) _(This makes it a good fit for cross thread communication and a cross network interface)_.
+
+At the high level protocol allows IPFS nodes to announce services they can provide to other nodes on the network, which can be requested by connected nodes to overcome their limitations.
```

Comment:
I feel the "Limited Relay" concept from  @aarshkshah1992's   Project Flare (#23) could fit here.
Worth mentioning it as potential use case.